client.go

```diff
@@ -127,6 +129,61 @@ type Client struct {
 	expectedAPIVersion  APIVersion
 }
 
+// NewClientFromEnv Returns a Client instance configured using
+// environment variables according to the same logic as the
+// official Docker client. It will use the latest remote API version
+// available in the server.
+func NewClientFromEnv() (*Client, error) {
+	client, err := NewVersionedClientFromEnv("")
+	if err != nil {
+		return nil, err
+	}
+	client.SkipServerVersionCheck = true
+	return client, nil
+}
+
+// NewVersionedClientFromEnv Returns a Client instance configured using
+// environment variables according to the same logic as the
+// official Docker client, using a specific remote API version.
+func NewVersionedClientFromEnv(apiVersionString string) (*Client, error) {
+	dockerHost := os.Getenv("DOCKER_HOST")
+	if dockerHost == "" {
+		return nil, errors.New("DOCKER_HOST must be set")
+	}
+	if os.Getenv("DOCKER_TLS_VERIFY") == "" {
+		return NewVersionedClient(dockerHost, apiVersionString)
+	}
+	parts := strings.SplitN(dockerHost, "://", 2)
```

Comment:
This may be redundant from https://github.com/fsouza/go-dockerclient/commit/f1f91d5ba55810454f1d75e61d61d8b4c45e6e9b